Инкремент 25

Fairness2 opened the pull request:
rpc и тесты

Fairness2 committed:
iter23.3 | Добавлены тесты для чтения ключей и конфигурации

В этом коммите добавлены тесты для проверки чтения публичных и приватных ключей из файлов, а также тесты для разных сценариев чтения конфигурационных файлов. Эти изменения позволят удостовериться в корректности работы функций по управлению конфигурацией и ключами.

Измененные файлы:
- internal/config/keys_test.go
- internal/config/types_test.go
- cmd/server/config/parse_test.go
- cmd/server/handlers/handlemetric/errors_test.go
- cmd/server/config/parse.go
- cmd/agent/collector/sender/sender.go
- cmd/agent/config/parse_test.go

Fairness2 committed:
`cleanup | Removed unused constant in test files`

Удалена неиспользуемая константа `malformedJson` в тестовых файлах для удобства чтения и уменьшения ненужного кода. Данное изменение не оказывает влияния на функциональность или результаты тестирования.

Измененные файлы:
- cmd/server/config/parse_test.go
- cmd/agent/config/parse_test.go

Fairness2 committed:
iter24 | Добавлен NetworkMiddleware и тесты, обновлен RestClient

Добавлен middleware для фильтрации запросов по подсети. Обновлен RestClient для использования реального IP-адреса, добавлены соответствующие тесты.

Список измененных файлов:
- internal/middlewares/network.go
- cmd/server/config/parse_test.go
- internal/middlewares/network_test.go
- cmd/agent/config/parse.go
- cmd/agent/sendpool/restclient.go
- cmd/server/config/fileconfig.go
- .gitignore
- cmd/server/main.go
- cmd/agent/sendpool/restclient_test.go
- cmd/server/config/parse.go
- cmd/agent/config/parse_test.go
- cmd/server/config/config.go

Fairness2 committed:
iter25 | Добавлена конфигурация сервера RPC и обновлено промежуточное сетевое программное обеспечение.

Добавлен новый адрес RPC-сервера и параметры командной строки для него. Обновлен NetworkMiddleware для поддержки фильтрации запросов по подсети как для HTTP, так и для gRPC.

Changed files:
- internal/middlewares/network.go
- internal/middlewares/network_test.go
- cmd/agent/sendpool/restclient.go
- cmd/server/config/parse.go
- cmd/server/config/config.go
- cmd/server/main_test.go
- cmd/agent/sendpool/mock_test.go
- go.mod
- cmd/agent/sendpool/pool_test.go
- go.sum

ekubyshin commented on internal/logger/middleware.go:
```diff
@@ -61,3 +64,24 @@ func (r *responseWriterWithLogging) WriteHeader(statusCode int) {
+		)
+	}()
+	resp, err = handler(ctx, req)
+	return resp, err
```
здесь можно сразу возвращать return handler(ctx, req)

Fairness2 replied:
Поправил

ekubyshin commented on internal/encrypt/decrypter.go:
```diff
@@ -70,6 +76,26 @@ func (d Decrypter) Middleware(next http.Handler) http.Handler {
+	if len(h) > 0 && d.privateKey != nil {
+		if r, isMR := req.(*pb.MetricsRequest); isMR {
+			decryptBody, err := d.decrypt(r.GetBody())
+			if err != nil {
```
в подобных конструкциях можно использовать guard pattern, инвертировав условие. Чтобы избавиться от такой большой вложенности условий

Fairness2 replied:
Немного развернул

ekubyshin commented on cmd/server/main.go:
```diff
@@ -100,8 +128,11 @@ func runApplication() error {
 	//cancel()
-	if err = stopServer(server, ctx); err != nil { // Запускаем сервер
-		return err
+	if err = stopServer(server, ctx); err != nil {
```
В целом тут все хорошо, но можно чуть улучшить обработку сигнала и errgroup. При обработке сигнала уже создается ctx, который можно везде использовать как родительский контекст. Далее использовать этот ctx при создании errgroup.WithContext, вернувшийся контекст можно дальше прокинуть и обраббатывать внутри wg.Go проверить отмену контекста.
Далее при создании в 99 строке, можно создавать контект с интервалом от родительского ctx, таким образом можно будет внутри синка отловить отмену контекста и тд.

Fairness2 replied:
Завязался на дочерний контекст

Fairness2 committed:
iter25.1 | Улучшена обработка контекста и упрощен код в различных модулях.

В `cmd/server/main.go` добавлено использование errgroup для контекста, что позволило улучшить работу с контекстами. В `internal/encrypt/decrypter.go` оптимизирована логика расшифровки тела запроса, а в тестах для базы данных `internal/database/pool_test.go` изменено тестовое значение dsn. Также в `internal/logger/middleware.go` устранен избыточный код.

Измененные файлы:
- cmd/server/main.go
- internal/encrypt/decrypter.go
- internal/database/pool_test.go
- internal/logger/middleware.go
